### PR TITLE
[BugFix] Support random distribution in dynamic partition scheduler 

### PR DESCRIPTION
**Pull request by LiShuMing**
Fixes #22401 
1. Support random distribution for aggregate key table;
2. Support random distribution in dynamic partition scheduler;

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 14 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/clone/DynamicPartitionScheduler.java|10|12|83.33%|[269, 270]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 18 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/clone/DynamicPartitionScheduler.java|10|12|83.33%|[269, 270]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|6|6|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java [hunk 1]
```diff
@@ -351,8 +354,10 @@ public static void analyze(CreateTableStmt statement, ConnectContext context) {
                     distributionDesc = new RandomDistributionDesc();
                 }
             }
-            if (distributionDesc instanceof RandomDistributionDesc && keysDesc.getKeysType() != KeysType.DUP_KEYS) {
-                throw new SemanticException("Random distribution must be used in DUP_KEYS", distributionDesc.getPos());
+            if (distributionDesc instanceof RandomDistributionDesc && keysDesc.getKeysType() != KeysType.DUP_KEYS 
+                    && keysDesc.getKeysType() != KeysType.AGG_KEYS) {
+                throw new SemanticException("Random distribution must be used in DUP_KEYS or AGG_KEYS",
+                        distributionDesc.getPos());
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:360]: agg without replace

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 18 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/clone/DynamicPartitionScheduler.java|10|12|83.33%|[269, 270]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 18 / 20 (90.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/clone/DynamicPartitionScheduler.java|10|12|83.33%|[269, 270]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|8|8|100.00%|[]|

**Review by meegoo — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by Astralidea** [comment]
https://github.com/Mergifyio backport branch-3.1

**Pull request merged**